rules/python/lang/weak_hash_md5.yml

```diff
@@ -1,58 +0,0 @@
-imports:
-  - python_shared_lang_datatype
-patterns:
-  - pattern: hashlib.md5($<OPTIONAL_DATA_TYPE>)
-    filters:
-      - either:
-          - variable: OPTIONAL_DATA_TYPE
-            detection: python_shared_lang_datatype
-            scope: result
-          - not:
-              variable: OPTIONAL_DATA_TYPE
-              detection: python_shared_lang_datatype
-              scope: result
-  - pattern: $<MD5_INIT>.update($<OPTIONAL_DATA_TYPE>)
-    filters:
-      - variable: MD5_INIT
-        detection: python_lang_weak_hash_md5_init
-        scope: cursor
-      - either:
-          - variable: OPTIONAL_DATA_TYPE
-            detection: python_shared_lang_datatype
-            scope: result
-          - not:
-              variable: OPTIONAL_DATA_TYPE
-              detection: python_shared_lang_datatype
-              scope: result
-auxiliary:
-  - id: python_lang_weak_hash_md5_init
-    patterns:
-      - hashlib.md5()
-languages:
-  - python
-skip_data_types:
-  - "Unique Identifier"
-  - "Passwords" # see python_lang_weak_password_encryption_md5
-metadata:
-  description: "Usage of weak hashing library (MD5)"
-  remediation_message: |-
-    ## Description
-
-    Using a weak hashing library, such as MD5, increases the risk of data breaches. MD5 itself is vulnerable to collision attacks, where two different inputs can produce the same hash value, compromising data integrity and security.
-
-    ## Remediations
-
-    - **Do not** use MD5 for hashing. This algorithm is considered weak and vulnerable to attacks.
-      ```python
-      hashlib.md5('password').digest() # unsafe
-      ```
-    - **Do** opt for stronger hashing algorithms like SHA-256 to enhance security.
-      ```python
-      hashlib.sha256('password').digest()
-      ```
-  cwe_id:
-    - 328
-  id: python_lang_weak_hash_md5
```

Comment:
What's the impact of deleting rules?

Reply:
I'm not really sure what expectations are pre-GA. I'll change it back